class-based-2/src/stack.ts

```diff
@@ -0,0 +1,21 @@
+import { App, Stack } from "aws-cdk-lib";
+import { Function, StepFunction } from "functionless";
+
+export class MyStack extends Stack {
```

Comment:
This is a nice hybrid between single file and CDK-leaning but it doesn't actually accomplish the objective of having functions in their own file. Showing it as another option to consider.